docs/SPIR-V.rst

```diff
@@ -365,6 +376,11 @@ Casting between (vectors) of scalar types is translated according to the followi
 |   Float    |                   | ``OpConvertFToS`` | ``OpConvertFToU`` |      no-op        |
 +------------+-------------------+-------------------+-------------------+-------------------+
 
+Indexing operator
++++++++++++++++++
+
+The ``[]`` operator can also be used to access elements in a matrix or vector. A matrix whose row and/or column count is 1 will be translated into a vector or scalar. If a variable is used as the index for the dimension whose count is 1, that variable will be ignored in the generated SPIR-V code. This is because out-of-bound indexing triggers undefined behavior anyway. For example, for a 1xN matrix ``mat``, ``mat[index][0]`` will be translated into ``OpAccessChain ... %mat %uint_0``. Similarly, variable index into a size 1 vector will also be ignored and the only element will be always returned.
+
```

Comment:
It may also be worth mentioning that:
`mat[i][j]` in HLSL refers to the element on row `i` and column `j` of the matrix, but will be translated to access chain into row `j` and column `i` of the SPIR-V matrix because the matrix is stored as column major (as explained in the `vectors and matrices` section above)